### PR TITLE
tests: General improvements to operator script

### PR DESCRIPTION
**Pull request by GabyCT**
This PR gives general improvements to the operator script that is being used in tests. This PR includes changes like better variable definition and the use of a variable that has been already defined.

### CODE REVIEW

**Review by fitzthum — APPROVED** [review]
LGTM

**Review by ldoktor — COMMENTED** [review]
Thanks, those are fine changes although shellcheck complains about more. Could you please take care of the others too since you are already treating style issues?

**Comment by GabyCT** [comment]
changes applied

File: tests/e2e/operator.sh [hunk 1]
```diff
@@ -247,8 +250,8 @@ uninstall_operator() {
 	cmd+="&& ! echo \$_OUT | grep -q -e cc-operator-controller-manager"
 	if ! wait_for_process 180 30 "$cmd"; then
 		echo "::error:: the controller manager is still running"
-
-		local pod_id="$(get_pods_regex $pod $op_ns)"
+		local pod_id
+		pod_id="$(get_pods_regex $pod $op_ns)"
```
Comment (ldoktor) [inline tests/e2e/operator.sh:254]: The `$pod` and `$op_ns` should be quoted as well

File: tests/e2e/operator.sh [hunk 2]
```diff
@@ -132,12 +135,12 @@ install_ccruntime() {
...
 	for pod in cc-operator-daemon-install cc-operator-pre-install-daemon; do
-		cmd="kubectl get pods -n "$op_ns" --no-headers |"
+		cmd="kubectl get pods -n $op_ns --no-headers |"
 		cmd+="egrep -q ${pod}.*'\<Running\>'"
 		if ! wait_for_process 600 30 "$cmd"; then
 			echo "::error:: $pod pod is not running"
-
-			local pod_id="$(get_pods_regex $pod $op_ns)"
+			local pod_id
+			pod_id="$(get_pods_regex $pod $op_ns)"
```
Comment (ldoktor) [inline tests/e2e/operator.sh:143]: The same here

File: tests/e2e/operator.sh [hunk 3]
```diff
@@ -101,12 +103,13 @@ install_operator() {
...
 	local controller_pod="cc-operator-controller-manager"
-	local cmd="kubectl get pods -n "$op_ns" --no-headers |"
+	local cmd="kubectl get pods -n $op_ns --no-headers |"
 	cmd+="egrep -q ${controller_pod}.*'\<Running\>'"
 	if ! wait_for_process 120 10 "$cmd"; then
 		echo "::error:: ${controller_pod} pod is not running"
 
-		local pod_id="$(get_pods_regex $controller_pod $op_ns)"
+		local pod_id
+		pod_id="$(get_pods_regex $controller_pod $op_ns)"
```
Comment (ldoktor) [inline tests/e2e/operator.sh:112]: and here

File: tests/e2e/operator.sh [hunk 4]
```diff
@@ -132,12 +135,12 @@ install_ccruntime() {
 	local pod=""
 	local cmd=""
 	for pod in cc-operator-daemon-install cc-operator-pre-install-daemon; do
-		cmd="kubectl get pods -n "$op_ns" --no-headers |"
+		cmd="kubectl get pods -n $op_ns --no-headers |"
```
Comment (ldoktor) [inline tests/e2e/operator.sh:138]: and here as well `cmd="kubectl get pods -n '$op_ns' --no-headers |"`

File: tests/e2e/operator.sh [hunk 5]
```diff
@@ -132,12 +135,12 @@ install_ccruntime() {
...
 		cmd+="egrep -q ${pod}.*'\<Running\>'"
```
Comment (ldoktor) [inline tests/e2e/operator.sh:139]: and here...

File: tests/e2e/operator.sh [hunk 6]
```diff
@@ -132,12 +135,12 @@ install_ccruntime() {
 	local pod=""
 	local cmd=""
 	for pod in cc-operator-daemon-install cc-operator-pre-install-daemon; do
-		cmd="kubectl get pods -n "$op_ns" --no-headers |"
+		cmd="kubectl get pods -n '$op_ns' --no-headers |"
```
Comment (ldoktor) [inline tests/e2e/operator.sh:138]: Thanks, this one is fine as it's a variable substitution, but ... (see below)

File: tests/e2e/operator.sh [hunk 7]
```diff
@@ -132,12 +135,12 @@ install_ccruntime() {
...
 		cmd+="egrep -q ${pod}.*'\<Running\>'"
 		if ! wait_for_process 600 30 "$cmd"; then
 			echo "::error:: $pod pod is not running"
-
-			local pod_id="$(get_pods_regex $pod $op_ns)"
+			local pod_id
+			pod_id="$(get_pods_regex '$pod' '$op_ns')"
```
Comment (ldoktor) [inline tests/e2e/operator.sh:143]: ^^ unlike variable substitution in shell execution the expansion starts over inside `$( ... )` therefore here you need to use `"` to expand the variables.

File: tests/e2e/operator.sh [hunk 8]
```diff
@@ -247,8 +250,8 @@ uninstall_operator() {
 	cmd+="&& ! echo \$_OUT | grep -q -e cc-operator-controller-manager"
 	if ! wait_for_process 180 30 "$cmd"; then
 		echo "::error:: the controller manager is still running"
-
-		local pod_id="$(get_pods_regex $pod $op_ns)"
+		local pod_id
+		pod_id="$(get_pods_regex '$pod' '$op_ns')"
```
Comment (ldoktor) [inline tests/e2e/operator.sh:254]: the same here, double quotes please

Comment (GabyCT) [reply tests/e2e/operator.sh]: @ldoktor thanks for the feedback changes applied

File: tests/e2e/operator.sh [hunk 9]
```diff
@@ -101,12 +103,13 @@ install_operator() {
...
 	local controller_pod="cc-operator-controller-manager"
-	local cmd="kubectl get pods -n "$op_ns" --no-headers |"
+	local cmd="kubectl get pods -n '$op_ns' --no-headers |"
 	cmd+="egrep -q ${controller_pod}.*'\<Running\>'"
 	if ! wait_for_process 120 10 "$cmd"; then
 		echo "::error:: ${controller_pod} pod is not running"
 
-		local pod_id="$(get_pods_regex $controller_pod $op_ns)"
+		local pod_id
+		pod_id="$(get_pods_regex '$controller_pod' '$op_ns')"
```
Comment (ldoktor) [inline tests/e2e/operator.sh:112]: This one still needs to be treated (`"` vs. `'`)

Comment (ldoktor) [reply tests/e2e/operator.sh:112]: (It should be the last thing, consider it acked once it is addressed - I'm on PTO already, not sure when I get to it)

Comment (GabyCT) [reply tests/e2e/operator.sh]: @ldoktor changes applied

**Review by ldoktor — APPROVED** [review]
Thanks, lgtm now :-) (sorry for the delay, PTO time...)

**Comment by ldoktor** [comment]
@stevenhorsman @wainersm the license check failed, do you have any experience with that? (this change should not have affected that)

**Comment by stevenhorsman** [comment]
> @stevenhorsman @wainersm the license check failed, do you have any experience with that? (this change should not have affected that)

I think we can ignore it as no-one in development has access to the report to check the issue. There seems to have been some sort of org setting that has triggered this check running on all PRs as we have the same issue in the CAA repo.

**Comment by ldoktor** [comment]
Well based on https://app.fossa.com/projects/git%2Bgithub.com%2Fconfidential-containers%2Foperator/refs/branch/main/abe928347a0ff9c3ecbeafceb0c761cc33cd4af1/browse/licenses I'd say it complains about `go.mongodb.org/mongo-driver` dependency, but that one is not new. Anyway I don't have access to the full report and also think it should be merged.

**Comment by fitzthum** [comment]
The license check has been failing for a while. Unfortunately it's really hard to get details on why. Nobody has an account with FOSSA, but sometimes you can stumble onto the issue on the FOSSA page. Last I checked it was actually complaining about Kata, but I'm not sure what is happening this time. I wouldn't let it block this PR.

**Comment by ldoktor** [comment]
Thanks for the confirmation, let me merge this awesome numbered PR :-)

**Pull request merged**